File: DesignPatternsExercises/java/StateDesignPattern/Exercise/CardInsertedState.java
public class CardInsertedState implements AtmMachineStates {
    @Override
    public boolean ejectCard() {
        System.out.println("Card is ejected...");
        return true;
    }

    @Override
    public boolean insertCard() {
        System.out.println("Card is already inserted into ATM Machine slot...");
        return false;
    }

    @Override
    public boolean withdraw() {
        System.out.println("Money are withdrawn...");
        return true;
    }
}

File: DesignPatternsExercises/ruby/state_design_pattern/exercise/card_inserted_state.rb
require_relative 'atm_machine_states'

class CardInsertedState < AtmMachineStates
  def __________(9)__________
    puts 'Card is ejected...'
    true
  end

  def insert_card?
    puts 'Card is already inserted into ATM Machine slot...'
    false
  end

  def withdraw?
    puts 'Money are withdrawn...'
    true
  end
end
